### PR TITLE
test: Added examples/reduced_bolero_test to demonstrate a problem I'm having

### PR DESCRIPTION
**Pull request by ckaran**
Complete information is in the README.md file, but the gist is that bolero
works correctly in test mode, but not in release mode.

### CODE REVIEW

**Comment by camshaft** [comment]
You've set `debug = false` and bolero, unfortunately, relies on those to be there to know if the test uses libtest or not. If there's a better mechanism to find that out, I'm open to it.

File: examples/reduced_bolero_test/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+debug-assertions = true
+codegen-units = 16
+incremental = true
+overflow-checks = true
+
+# The benchmarking profile, used for `cargo bench` (and the test targets and
+# unit tests for `cargo test --release`).
+[profile.bench]
+opt-level = 3
+debug = false
```
Comment (camshaft) [inline examples/reduced_bolero_test/Cargo.toml:80]: Is your issue fixed when setting this to `true`?

Comment (camshaft) [reply examples/reduced_bolero_test/Cargo.toml:80]: I was just able to test this and, no, it's not fixed ☹️

**Comment by camshaft** [comment]
Fixed in #17

**Comment by ckaran** [comment]
I *think* that this fixed the problem of not finding the tests, but I may have turned up a new issue.  Basically, there are some tests that return `test XXX has been running for over 60 seconds` on the stable channel which fail on the nightly channel.

I'm testing the Cartesian product of `{stable, beta, nightly} X {debug, release}` right now, and I'm going to see which tests fail.  If I can find one particular test that is consistent in doing this, I'll try to reduce it down to a new test and create a PR for it.  This will likely take a couple of days, but if you can think of any reason why bolero would time out on stable, but fail on nightly (and, to be clear, 'fail' here means that an assert statement within `fuzz!()` has been tripped), then maybe you can start working on that while I work on a reduced test case.

**Pull request closed without merging**